apps/deep-clone-main/src/locations/Sidebar.tsx

```diff
@@ -1,55 +1,90 @@
-import { useState } from 'react';
-import { Stack, Button, Caption } from '@contentful/f36-components';
-import { useSDK } from '@contentful/react-apps-toolkit';
+import { useEffect, useState } from 'react';
+import { Text, Button, Stack } from '@contentful/f36-components';
+import { useAutoResizer, useSDK } from '@contentful/react-apps-toolkit';
 import { SidebarAppSDK } from '@contentful/app-sdk';
 import EntryCloner from '../utils/EntryCloner';
 import { AppParameters } from '@/vite-env';
 
 function Sidebar() {
+  const REDIRECT_DELAY = 3000;
   const sdk = useSDK() as SidebarAppSDK;
+  useAutoResizer();
 
-  const installationParams: AppParameters = {
-    cloneText: 'Copy',
-    cloneTextBefore: true,
-    automaticRedirect: true,
-    ...sdk.parameters.installation,
-  };
+  const parameters = sdk.parameters.installation as AppParameters;
 
   const [isLoading, setLoading] = useState<boolean>(false);
   const [isDisabled, setDisabled] = useState<boolean>(false);
+  const [finished, setFinished] = useState<boolean>(false);
+  const [referencesCount, setReferencesCount] = useState<number>(0);
+  const [clonesCount, setClonesCount] = useState<number>(0);
+  const [updatesCount, setUpdatesCount] = useState<number>(0);
+  const [countdown, setCountdown] = useState<number>(REDIRECT_DELAY / 1000);
+
+  useEffect(() => {
+    if (finished && parameters.redirect) {
+      const interval = setInterval(() => {
+        if (countdown > 0) {
+          setCountdown((currentCountdown) => currentCountdown - 1);
+        } else {
+          clearInterval(interval);
+        }
+      }, 1000);
+      return () => clearInterval(interval);
+    }
+    return undefined;
+  }, [finished]);
```

Comment:
Perhaps we can avoid using the finished state. Coundown starts in 0
```suggestion
useEffect(() => {
  if (countdown === 0) return;

  const interval = setInterval(() => {
    setCountdown((currentCountdown) => {
      if (currentCountdown <= 1) {
        clearInterval(interval);
        return 0;
      }
      return currentCountdown - 1;
    });
  }, 1000);

  return () => clearInterval(interval);
}, [countdown]);
```